src/ui/Analytics/Analytics.ts

```diff
@@ -463,6 +492,31 @@ export class Analytics extends Component {
     );
   }
 
+  private isGtmScriptPresent() {
+    return _.some(document.querySelectorAll('script'), scriptElement => {
+      const regex = new RegExp(`www\.googletagmanager\.com`);
+      return regex.test(scriptElement.innerText) || regex.test(scriptElement.src);
+    });
+  }
+
+  private handleAnalyticsEventReady(data: IAnalyticsEventArgs) {
+    if (!this.options.pushToGtmDataLayer) {
+      return;
+    }
+
+    const dataLayerName = this.options.gtmDataLayerName;
+
+    if (!(<any>window)[dataLayerName]) {
+      (<any>window)[dataLayerName] = [];
+    } else if (
+      typeof (<any>window)[dataLayerName] === 'object' &&
+      typeof (<any>window)[dataLayerName].push === 'function' &&
+      (<any>window)[dataLayerName].push !== Array.prototype.push
```

Comment:
I think only checking for the presence of `window[dataLayerName]` should be sufficient.

And I'd wrap the call to `window[dataLayerName].push` to a try catch, which simply logs the error in an explicit manner in the console, but does not block the rest of the execution (eg: "normal" coveo analytics call)